lang/cs/Org.Apache.REEF.Tests/Functional/IMRU/TestTaskExceptions.cs

```diff
@@ -0,0 +1,251 @@
+        {
+            private readonly IEvaluatorRequestor _requestor;
+            private readonly IConfiguration _taskConfiguration;
+
+            [Inject]
+            private FailedTaskDriver(
+                IEvaluatorRequestor requestor,
+                [Parameter(typeof(TaskConfigurationString))] string taskConfigString,
+                AvroConfigurationSerializer avroConfigurationSerializer)
+            {
+                _requestor = requestor;
+                _taskConfiguration = avroConfigurationSerializer.FromString(taskConfigString);
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitTask(_taskConfiguration);
+            }
+
+            /// <summary>
+            /// Verify Exception message and exception type for different task id. 
+            /// </summary>
+            /// <param name="value"></param>
+            public void OnNext(IFailedTask value)
+            {
+                var msg = string.Format(CultureInfo.InvariantCulture,
+                    "In IFailedTask, taskId: {0}, Message {1}, Exception {2}.",
+                    value.Id,
+                    value.Message,
+                    value.AsError().GetType());
+
+                Logger.Log(Level.Info, msg);
+                if (value.Id.Equals(TaskId + 1))
+                {
+                    Assert.Equal(TaskManager.TaskAppError, value.Message);
+                    if (value.AsError() == null || !(value.AsError() is IMRUTaskAppException))
+                    {
+                        throw new Exception("Exception should have been serialized properly.");
+                    }
+                }
+
+                if (value.Id.Equals(TaskId + 2))
+                {
+                    Assert.Equal(TaskManager.TaskGroupCommunicationError, value.Message);
+
+                    if (value.AsError() == null || !(value.AsError() is IMRUTaskGroupCommunicationException))
+                    {
+                        throw new Exception("Exception should have been serialized properly.");
+                    }
+                }
+
+                if (value.Id.Equals(TaskId + 3))
+                {
+                    Assert.Equal(TaskManager.TaskSystemError, value.Message);
+                    if (value.AsError() == null || !(value.AsError() is IMRUTaskSystemException))
+                    {
+                        throw new Exception("Exception should have been serialized properly.");
+                    }
+                }
+                Logger.Log(Level.Info, ValidFailedTaskMessage);
+                value.GetActiveContext().Value.Dispose();
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                throw new Exception("Did not expect a completed task.");
```

Comment:
No. This exception is thrown in driver, not in test. So it gets logged in driver.stdout, and the test continues.